js/classes/game_engine.js

```diff
@@ -74,22 +74,28 @@ class GameEngine {
      * This method is going to go through all entities and allow them to update their position.
      */
     update() {
-        // Iterate forward through the entities array. Update everything.
-        for (let i = 0; i < this.entities.length; i++) {
-            let entity = this.entities[i];
-            if (!entity.removeFromWorld) {
-                entity.update();
+        // Update the HUD and Crosshair regardless of whether the game is running or not
+        HUD.update();
+        CROSSHAIR.update();
+
+        if (this.running) {
+            // Iterate forward through the entities array. Update everything.
+            for (let i = 0; i < this.entities.length; i++) {
+                let entity = this.entities[i];
+                if (!entity.removeFromWorld) {
+                    entity.update();
+                }
             }
-        }
-        // Update Chad, who is not a regular entity.
-        CHAD.update();
-        // Update the camera, which is not a regular entity.
-        CAMERA.update();
+            // Update Chad, who is not a regular entity.
+            CHAD.update();
+            // Update the camera, which is not a regular entity.
+            CAMERA.update();
 
-        // Iterate backward through the entities array, and remove all entities which ought be removed.
-        for (let i = this.entities.length - 1; i >= 0; i--) {
-            if (this.entities[i].removeFromWorld) {
-                this.entities.splice(i, 1);
+            // Iterate backward through the entities array, and remove all entities which ought be removed.
+            for (let i = this.entities.length - 1; i >= 0; i--) {
```

Comment:
Why iterate backwards? Not sure it makes a difference, just curious if there is a reason. Is it because of how splice works?

Reply:
Yes, that's exactly right. If you iterate forwards while removing items, you'll end up skipping some.